app/Libraries/OsuAuthorize.php

```diff
@@ -712,7 +724,18 @@ public function checkBeatmapsetMetadataEdit(?User $user, Beatmapset $beatmapset)
             return 'ok';
         }
 
-        if ($user->getKey() === $beatmapset->user_id && in_array($beatmapset->approved, $ownerEditable, true)) {
+        if ($user->isProjectLoved() && $beatmapset->isLoved()) {
+            return 'ok';
+        }
+
+        if ($user->isBNG() && in_array($beatmapset->approved, $bnEditable, true)) {
+            return 'ok';
+        }
+
+        if ($user->getKey() === $beatmapset->user_id &&
+            in_array($beatmapset->approved, $ownerEditable, true) &&
+            !$beatmapset->hasNominations()
+        ) {
```

Comment:
This check should be changed so the user gets a useful error message when the beatmapset has nominations instead of the generic unauthorized error.